### PR TITLE
Edit view columns

### LINKED ISSUE #473: Edit view columns (all available types)
## Description
A user should be able to add and remove columns from the view grid, as well as rename existing columns. When adding a new column, they should be guided through the process of picking a column type and filling out the necessary settings.

## Relevant Job Stories
* When I'm building my view or working with it later, I want to be able to add columns so that I can display and work with only the data that I need for the task at hand
* When I'm adding a column, I want to be able to name it so that the resulting view communicates the content and purpose of the column
* When I'm adding a column, I want it to automatically populate with data from Zetkin so that everything stays in sync
* When I'm adding a column, I want to be able to choose from one of several types so that I can work with any type of data from Zetkin in the view. These are the options currently:
* When I'm choosing the type of column, I want to be able to choose
  * …a field column that renders a field from a person record, so that I can see things like names, phone numbers and addresses in views
  * …a Smart Search query cross-reference column and pick a Smart Search, so that I can see whether each person in the view also matches that Smart Search query
  * …a survey response column and pick an existing survey and question, so that I can see for every person what they have responded (if anything) to that question
  * …a survey submit date column and pick a survey, so that I can see for every person what date they submitted that survey (if at all)
  * …a "notes" column, so that I can see the most recent note on each person
  * …a tag column and pick a tag, so I can toggle that tag on/off for people in the view individually
  * …a "toggle" column that I can toggle on/off for each person
  * …a person column, so that I can assign and reassign an organizer to each person record in the view
* When I don't need a column, I want to be able to delete it so that I can free upp space in the view and display only the data that I need for the task at hand

## Prerequisites
This task can be started without prerequisites.

## Requirements
- [x] To the right of the last column should be an "add column" button
  - [x] Clicking "add column" should open a big modal (referred to as "Column Gallery"), similar to the Smart Search edit modal
  - [x] At first, the modal should present all the different column types (see job stories above)
  - [x] Picking a column type should proceed to a second step, where the user can configure the column
  - [x] After editing, the user should be allowed to pick a name
  - [x] A default name should be already filled in, based on column settings
  - [ ] If adding the column fails, an error message should be displayed
- [x] The column menu should have an "Edit settings" option, which opens the Column Gallery (straight to step 2 for settings)
- [x] The column menu should have a "Rename" option
  - [x] Clicking the rename option should open a modal where user can change the name
- [x] The column menu should have a "Remove from view" option
  - [ ] If the column is a "local" column, a modal should appear to confirm deleting it (as data will be lost)

## Possible implementations
Look at the Smart Search modal (and it's "Filter Gallery") for how to implement the "Column Gallery".

The "Add column" button in the column header can be implemented by adding another `GridColDef` to `columns` and rendering a button in the header (using `renderHeader`). I don't know if that's a good solution, but it seems to work.

Relevant API resources:
* Create columns using `POST /orgs/{org_id}/people/views/{view_id}/columns`
* Edit a column using `PATCH /orgs/{org_id}/people/views/{view_id}/columns/{col_id}`
* Delete a column using `DELETE /orgs/{org_id}/people/views/{view_id}/columns/{col_id}`

## Design specifications
There are no full wireframes for this flow. The wireframes below show the "Add" button on the right hand side in the table header. 

image

## Open questions
What should this look like exactly? Some developer creativity needed here.

### PR DESCRIPTION
**Pull request by richardolsson**
This PR adds interface to create, edit and remove columns from a view.

Reworks CI to reduce number of Playwright workers to 4, removes caching, and only installs chromium.

Closes #473

### CODE REVIEW

**Comment by richardolsson** [comment]
Left to do (handing over to @djbusstop for today):

- [ ] Improve Typescript types for columns
   - [ ] Look at Smart Search for states (New*, *WithId etc)
   - [ ] Create discreet types for columns (by type)
- [ ] Should `title` be in the config dialog?
   - [ ] Remove title input and auto-generate title
- [ ] Styling of big dialog
   - [ ] Titles and buttons on cards
   - [ ] Make smaller
   - [ ] Use SubmitCancel buttons
   - [ ] Cancel should go back to type select
   - [ ] Config forms
     - [ ] Add labels
     - [ ] Vertical layout
     - [ ] Centered in dialog
- [ ] Error messages
- [ ] Add tests
   - [ ] Integration test creating one column type
   - [ ] Integration test that deleting column sends correct request
   - [ ] Integration test that renaming column sends correct request
   - [ ] Unit test the rest (if not too time consuming)

File: playwright/tests/organize/people/views/columns/delete-column.spec.ts
Comment (richardolsson) [inline playwright/tests/organize/people/views/columns/delete-column.spec.ts:61]: Copy-paste mistake? Variable should be called `columnDelReq` IMO.

File: playwright/tests/organize/people/views/columns/rename-column.spec.ts
Comment (richardolsson) [inline playwright/tests/organize/people/views/columns/rename-column.spec.ts:71]: Same copy-paste mistake here? 😊

File: src/components/views/ViewDataTable/ViewDataTableToolbar.tsx
Comment (richardolsson) [inline src/components/views/ViewDataTable/ViewDataTableToolbar.tsx:12]: Why the empty box? Is it just for the layout? Then maybe `justifyContent="flex-end"` could be an alternative?

File: src/components/views/ViewDataTable/ViewDataTableColumnMenu.tsx
Comment (richardolsson) [inline src/components/views/ViewDataTable/ViewDataTableColumnMenu.tsx:43]: Do these actually have to be unique? Including the column ID requires the test to know it (which is fine, because it's mocked, but does mean the same data is stored in multiple places). I thought about it when I read the test, but at the time imagined the test ID to be coming from inside MUI (for some reason, obviously unreasonable). Now that I see it's in our control, I'm thinking we could make it only locally unique (without the suffix) and use descendent selectors in the tests. What do you think?

**Review by richardolsson — COMMENTED** [review]
Awesome working bringing this home!

I can't actually formally approve this because I opened the PR myself, but this is my informal approval. 😊 

I did find some minor things that I'm commenting on below. But none of them are blockers. So I'll leave it up to you @djbusstop to decide if you want to approve right away or make (some of) those changes first.

File: playwright/tests/organize/people/views/columns/delete-column.spec.ts
Comment (djbusstop) [reply playwright/tests/organize/people/views/columns/delete-column.spec.ts:61]: Thanks for catching these. Shows you really read it!

**Comment by richardolsson** [comment]
Actually I found an issue. I can't reconfigure a tag column to use another tag. I get a 400 error saying "Invalid data" (which is a really bad error message). Looking at the platform code, my guess is that it's because the front-end is sending `description: null`. The description should either be a string or omitted entirely from the object, i.e. not `null`.

Can you see if you can reproduce and maybe fix before merge, @djbusstop?

**Comment by djbusstop** [comment]
@richardolsson i pushed all requested changes and bug fixes. also, love how consistently and quickly tests pass now. I guess... I review and approve?

**Review by djbusstop — APPROVED** [review]
great work, us

**Pull request merged**
